File: Leetcode/JS/Leetcode384-打乱数组.js
var Solution = function(nums) {
    this.memo = nums;
};

/**
 * @return {number[]}
 */
Solution.prototype.reset = function() {
    return this.memo;
};

/**
 * @return {number[]}
 */
Solution.prototype.shuffle = function() {
    let cp = this.memo.slice();
    let n = cp.length;
    for(let i = 0;i < n;i++){
        let  t = Math.floor(Math.random() * 200 + 1) % (n - i);
        [cp[i],cp[i + t]] = [cp[i + t],cp[i]];
    }
    return cp;
};

File: Leetcode/PY/Leetcode384-打乱数组.py
class Solution:

    def __init__(self, nums: List[int]):
        self.memo = nums

    def reset(self) -> List[int]:
        return self.memo

    def shuffle(self) -> List[int]:
        cp = copy.deepcopy(self.memo)
        shuffle(cp)
        return cp
